provision/acc_provision/apic_provision.py

```diff
@@ -1431,6 +1431,37 @@ def cloudfoundry_tn(self):
                    ["app_profile"])
         app_epg_name = (
             self.config["cf_config"]["default_endpoint_group"]["group"])
+        custom_epg_name = (
+            self.config["aci_config"]["cloudfoundry"]["custom_epg"])
+        isolated_segment = (
+            self.config["aci_config"]["cloudfoundry"]["isolated_segment"])
+        children_epg_objs, children_apic_objects = [], []
+
+        if isolated_segment:
+            isolated_subnet = (
+            self.config["aci_config"]["cloudfoundry"]["isolated_subnet"])
+            node_subnet.append(isolated_subnet)
+
+            is_gorouter_contract = self.contract(
+            'is-gorouter',
+            subjects=[dict(name='is-gorouter-subj', filters=['tcp-all'])])
+
+            isolated_node_epg = self.epg(
+            "cf-node-isolated",
+            "cf-node-bd",
+            provides=["is-node-all"],
+            consumes=["is-node-all", "is-gorouter", "%s-l3out-allow-all" % system_id],
+            vmm_domains=[(nvmm_type, nvmm_name)])
+
+            isolated_app_epg = self.epg(
+            "cf-isolated-app",
```

Comment:
the name of the app-epg for isolation segment is the same as the name of the isolation segment (and that name needs to be a config parameter)